orchagent/copporch.cpp

```diff
@@ -0,0 +1,474 @@
+            {
+                attr.id = SAI_POLICER_ATTR_PBS;
+                attr.value.u64 = std::stoul(fvValue(*i));
+                policer_attribs.push_back(attr);
+                SWSS_LOG_DEBUG("obtained pbs:%d", attr.value.u64);
+            }
+            else if (fvField(*i) == copp_policer_pir_field)          
+            {
+                attr.id = SAI_POLICER_ATTR_PIR;
+                attr.value.u64 = std::stoul(fvValue(*i));
+                policer_attribs.push_back(attr);
+                SWSS_LOG_DEBUG("obtained pir:%d", attr.value.u64);
+            }
+            else if (fvField(*i) == copp_policer_action_green_field)
+            {
+                SWSS_LOG_DEBUG("green action:%s", queue_ind.c_str());
+                sai_packet_action_t policer_action = packet_action_map.at(fvValue(*i));
+                attr.id = SAI_POLICER_ATTR_GREEN_PACKET_ACTION;
+                attr.value.s32 = policer_action;
+                policer_attribs.push_back(attr);
+            }
+            else if (fvField(*i) == copp_policer_action_red_field) 
+            {
+                SWSS_LOG_DEBUG("red action:%s", queue_ind.c_str());
+                sai_packet_action_t policer_action = packet_action_map.at(fvValue(*i));
+                attr.id = SAI_POLICER_ATTR_RED_PACKET_ACTION;
+                attr.value.s32 = policer_action;
+                policer_attribs.push_back(attr);
+            }
+            else if (fvField(*i) == copp_policer_action_yellow_field)
+            {
+                SWSS_LOG_DEBUG("yellowaction:%s", queue_ind.c_str());
+                sai_packet_action_t policer_action = packet_action_map.at(fvValue(*i));
+                attr.id = SAI_POLICER_ATTR_YELLOW_PACKET_ACTION;
+                attr.value.s32 = policer_action;
+                policer_attribs.push_back(attr);
+            }
+            else
+            {
+                SWSS_LOG_ERROR("Unknown copp field specified:%s\n", fvField(*i).c_str());
+                return task_process_status::task_invalid_entry;
+            }
+        }
+
+        if (m_trap_group_map.find(key) != m_trap_group_map.end())
+        {
+            SWSS_LOG_DEBUG("found existing trap_group object:%s", key.c_str());
+            if (!policer_attribs.empty())
+            {
+                sai_object_id_t policer_id = getPolicer(key);
+                if (SAI_NULL_OBJECT_ID == policer_id)
+                {
+                    SWSS_LOG_WARN("Creating policer for existing Trap group:%llx (name:%s).", m_trap_group_map[key], key.c_str());
+                    if (!createPolicer(key, policer_attribs))
+                    {
+                        return task_process_status::task_failed;
+                    }
+                    SWSS_LOG_DEBUG("Created policer:%llx for existing trap group", policer_id);
+                }
+                else 
+                {
+                    for(sai_uint32_t ind = 0; ind < policer_attribs.size(); ind++)
```

Comment:
>or(sai_uint32_t ind = 0; ind < policer_attribs.size(); ind++) [](start = 21, length = 61)

add a TODO comment here.

/* TODO: We should really only set changed attributes. The changes need to detected either by orch agent submodule or by the orch agent framework. */